xarray/backends/zarr.py

```diff
@@ -311,13 +278,21 @@ class ZarrStore(AbstractWritableDataStore):
     """Store for reading and writing data via zarr
     """
 
-    def __init__(self, store=None, mode='a', synchronizer=None, group=None,
+    @classmethod
+    def open_group(cls, store, mode='r', synchronizer=None, group=None,
+                   writer=None):
+        import zarr
+        zarr_group = zarr.open_group(store=store, mode=mode,
+                                     synchronizer=synchronizer, path=group)
+        return cls(zarr_group, mode=mode, synchronizer=synchronizer,
+                   group=group, writer=writer)
+
+    def __init__(self, zarr_group, mode='r', synchronizer=None, group=None,
```

Comment:
I think all of these attributes (`mode`, `synchronizer` and `group`) can be pulled out of the zarr Group object. So can we drop them from the constructor?